movie_search/media.py

```diff
@@ -1,113 +1,265 @@
-        "provider_id", flat=True
-    )
-    return providers
-
-def get_movie_discover_data(
-     genres, person_id, sort_options, region, watch_region, providers, year
-):
-    return tmdb_api.get_data_from_endpoint(
-        "/discover/movie",
-        region=region,
-        primary_release_year=year,
-        with_genres=list(genres),
-        sort_by=sort_options,
-        watch_region=watch_region,
-        with_watch_providers=list(providers),
-        with_people=person_id,
-    )
+        if genres is not None:
+            tv_genres = store_genres(genres)
+
+        tv.genres.add(*tv_genres)
+
+        tv_recommendations = store_recommendations(tv.tmdb_id)
+        tv.recommendation.add(*tv_recommendations)
+
+        videos = self.store_videos(tv, video_data)
+
+        return tv, videos
+
+    def store_genres(self, genres):
+        tv_genres = []
+
+        for row in genres:
+            genre, created = TVSeriesGenre.objects.get_or_create(
+                name=row.get("name", ""),
+                genre_id=row.get("id", 0),
+            )
+            tv_genres.append(genre)
+
+        return tv_genres
+
+    def store_videos(self, tv_obj, video_data):
+        for video in video_data.get("results", []):
+            TVSeriesVideo.objects.get_or_create(
+                tv=movie_obj,
+                name=video_data.get("name", ""),
+                key=video_data.get("key", ""),
+            )
+        return TVSeriesVideo.objects.filter(tv_id=tv_obj.id)
+
+    def store_recommendations(self, tmdb_id):
+        recommendations_data = tmdb_api_obj.get_data_from_endpoint(
+            f"/tv/{tmdb_id}/recommendations"
+        )
+        tv_recommendations = []
+
+        for rec_data in recommendations_data.get("results", []):
+            recommendation, created = TVSeriesRecommendation.objects.get_or_create(
+                tmdb_id=rec_data.get("id", 0),
+                poster_path=rec_data.get("poster_path", ""),
+            )
+            movie_recommendations.append(recommendation)
+        return movie_recommendations
+
+    def get_genres_from_discover(self, genre_names):
+        genres = TVSeriesGenre.objects.filter(name__in=genre_names).values_list(
+            "genre_id", flat=True
+        )
+        return genres
+
+    def get_providers_from_discover(self, watch_provider_names):
+        providers = TVSeriesProvider.objects.filter(
+            name__in=watch_provider_names
+        ).values_list("provider_id", flat=True)
+        return providers
+
+    def get_discover_data(
+        self,
+        language,
+        year,
+        genres,
+        sort_options,
+        providers,
+    ):
+        return tmdb_api_obj.get_data_from_endpoint(
+            "/discover/movie",
+            language=language,
+            first_air_date_year=year,
+            with_genres=list(genres),
+            sort_by=sort_options,
+            with_watch_providers=list(providers),
+        )
+
+class PersonStrategy:
```

Comment:
the methods in this class merely call other methods, what is the goal of wrapping them in this class?

Reply:
Replied on this in Circle since I have a general convo there going about the classes used overall.

Reply:
Working on removing the strategy pattern, but see recent circle comments on this.

Reply:
It was a good experiment, but it has added more code + complexity. I am still a bit skeptical we win much from this pattern, because of: 

<img width="752" alt="image" src="https://github.com/vnoelifant/movie-search-app/assets/387927/521b3fad-05f4-426d-94aa-cbb6eaa4fae4">

Reply:
So key question: is there a need or use case here of "dynamically changing the behavior"?

(Better to discuss here btw = closer to the code)